app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+cache_archive = {}
```

Comment:
move it into the decorator. then for every decorated function will be created a new dictionary to store it results